### PR TITLE
Bumps collector version to v0.30.0

### PR DESCRIPTION
**Pull request by VineethReddy02**
fixes: #339

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 Changes by Version
 ==================
 
+0.30.0 (2021-07-15)
+-------------------
+* Bumped OpenTelemetry Collector to v0.30.0
+* Add CI check for auto-generated files ([#336](https://github.com/open-telemetry/opentelemetry-operator/pull/336), [@VineethReddy02](https://github.com/VineethReddy02))
```
Comment (jpkrohling) [inline CHANGELOG.md:7]: I think this should be left out. The changelog should contain only entries relevant to our end users.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+* Container Security Context ([#332](https://github.com/open-telemetry/opentelemetry-operator/pull/332), [@owais](https://github.com/owais))
+* Small manual adjusts for docs and bundle for v0.29.0 ([#329](https://github.com/open-telemetry/opentelemetry-operator/pull/329), [@jpkrohling](https://github.com/jpkrohling))
```
Comment (jpkrohling) [inline CHANGELOG.md:9]: This should also be out.

**Review by jpkrohling — COMMENTED** [review]
What do you think about turning this type of commit into release ones? As such, they would have to follow the first steps from the RELEASE.md

Concretely, this PR would be missing:

1. bundle changes
2. update the compat matrix

This way, we could release the operator right after this type of commit.

**Comment by VineethReddy02** [comment]
@jpkrohling I haven't noticed any migration-related steps in OtelCol changelog. But I would like you to confirm this :)

**Comment by jpkrohling** [comment]
Looks good, can you just confirm that you ran the `make bundle` as described in the readme, instead of just bumping the versions?

**Comment by VineethReddy02** [comment]
yup, we do have checks for the bundle if any changes are left locally as we recently added `ensure-generate-is-noop` in Makefile.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**